src/pool/srv_pool.c

```diff
@@ -3775,6 +3775,175 @@ pool_find_all_targets_by_addr(uuid_t pool_uuid,
+	 * need to expand the array
+	 */
+	if (arg->next_idx == *arg->nhandles) {
+		*arg->nhandles += 1;
+		D_REALLOC_ARRAY(*arg->handles_uuids, *arg->handles_uuids,
+				*arg->nhandles);
+		if (*arg->handles_uuids == NULL)
+			return -DER_NOMEM;
+	}
+	D_ASSERT(arg->next_idx < *arg->nhandles);
+
+	/* Store the uuid */
+	uuid_copy((*arg->handles_uuids)[arg->next_idx], *uuid);
+	arg->next_idx++;
+
+	return rc;
+}
+
+static int
+redist_open_hdls_send_rpcs(uuid_t pool_uuid, uuid_t *handles_uuids,
+			   uint32_t nhandles, d_rank_list_t *targets)
+{
+	crt_rpc_t			*tf_req;
+	struct pool_tgt_fetch_hdls_in	*tf_in;
+	struct pool_tgt_fetch_hdls_out	*tf_out;
+	int				topo;
+	int				rc;
+
+	/* Collective RPC to destroy the pool on all of targets */
+	topo = crt_tree_topo(CRT_TREE_KNOMIAL, 4);
+	rc = crt_corpc_req_create(dss_get_module_info()->dmi_ctx, NULL,
+				  targets, POOL_TGT_FETCH_HDLS, NULL, NULL,
+				  0, topo, &tf_req);
+	if (rc)
+		return rc;
+
+	tf_in = crt_req_get(tf_req);
+	D_ASSERT(tf_in != NULL);
+	uuid_copy(tf_in->tfi_pool_uuid, pool_uuid);
+	d_iov_set(&tf_in->tfi_hdls, handles_uuids, sizeof(uuid_t) * nhandles);
+
+	rc = dss_rpc_send(tf_req);
+	if (rc == 0 && DAOS_FAIL_CHECK(DAOS_POOL_DESTROY_FAIL_CORPC))
+		rc = -DER_TIMEDOUT;
+	if (rc != 0)
+		D_GOTO(out_rpc, rc);
+
+	tf_out = crt_reply_get(tf_req);
+	rc = tf_out->tfo_rc;
+	if (rc != 0)
+		D_ERROR(DF_UUID": failed to send handle uuids to targets "
+			DF_RC"\n", DP_UUID(pool_uuid), DP_RC(rc));
+out_rpc:
+	crt_req_decref(tf_req);
+
+	return rc;
+}
+
+static int
+redist_open_hdls(uuid_t pool_uuid, d_rank_list_t *targets)
+{
+	struct pool_svc			*svc;
+	struct redist_open_hdls_arg	 arg;
+	uint32_t			 connectable;
+	struct rdb_tx			 tx;
+	d_iov_t				 value;
+	uuid_t				*handles_uuids;
+	uint32_t			 nhandles;
+	int				 rc;
+
+	rc = pool_svc_lookup_leader(pool_uuid, &svc, NULL /* hint */);
+	if (rc != 0)
+		return rc;
+
+	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
+	if (rc != 0)
+		D_GOTO(out_svc, rc);
+
+	ABT_rwlock_rdlock(svc->ps_lock);
+
+	/* Check if pool is being destroyed and not accepting connections */
+	d_iov_set(&value, &connectable, sizeof(connectable));
+	rc = rdb_tx_lookup(&tx, &svc->ps_root,
+			   &ds_pool_prop_connectable, &value);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+	D_DEBUG(DF_DSMS, DF_UUID": connectable=%u\n",
+		DP_UUID(pool_uuid), connectable);
+	if (!connectable) {
+		D_ERROR(DF_UUID": being destroyed, not accepting connections\n",
+			DP_UUID(pool_uuid));
+		D_GOTO(out_lock, rc = -DER_BUSY);
+	}
+
+	/* Check how many handles are currently open */
+	d_iov_set(&value, &nhandles, sizeof(nhandles));
+	rc = rdb_tx_lookup(&tx, &svc->ps_root, &ds_pool_prop_nhandles, &value);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+
+	/* Abort early if there are no open handles */
+	if (nhandles == 0)
+		D_GOTO(out_lock, rc);
+
+	D_ALLOC_ARRAY(handles_uuids, nhandles);
+	if (handles_uuids == NULL)
+		D_GOTO(out_lock, rc=-DER_NOMEM);
```

Comment:
(style)  spaces required around that '=' (ctx:VxO)
(style)  space required before that '-' (ctx:OxV)